synapse/storage/databases/main/event_federation.py

```diff
@@ -726,17 +726,39 @@ def _get_auth_chain_difference_txn(
     async def get_backfill_points_in_room(
         self,
         room_id: str,
+        current_depth: int,
+        limit: int,
     ) -> List[Tuple[str, int]]:
         """
         Gets the oldest events(backwards extremities) in the room along with the
-        approximate depth. Sorted by depth, highest to lowest (descending).
+        approximate depth. Sorted by depth, highest to lowest (descending) so the closest
+        events to the `current_depth` are first in the list.
+
+        We use this function so that we can compare and see if a client's
+        `current_depth` at their current scrollback is within pagination range
+        of the event extremities. If the `current_depth` is close to the depth
+        of given oldest event, we can trigger a backfill.
+
+        We ignore extremities that have a greater depth than our `current_depth`
+        as:
```

Comment:
```suggestion
        We ignore extremities that are later than the user's current
        scroll position (ie, those with depth greater than `current_depth`) as:
```

Reply:
This sounds good but I feel like "later" is ambiguous when talking about a point in this list.

"extremities that are later" could be interpreted as older than the `current_depth` but it's actually the opposite. We ignore anything newer than the `current_depth`

 - oldest vs latest
 - earlier vs later
 - older vs newer

I've updated it to say "newer". Feel free to suggest another iteration